flake.nix

```diff
@@ -0,0 +1,162 @@
+{
+  description = "bitcoinj";
+
+  # this allows derivations with `__noChroot = true` and allows us to work around limitations with gradle
+  # see https://zimbatm.com/notes/nix-packaging-the-heretic-way
+  nixConfig.sandbox = "relaxed";
+
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
+
+    flake-parts = {
+      url = "github:hercules-ci/flake-parts";
+      inputs.nixpkgs-lib.follows = "nixpkgs";
+    };
+
+    gitignore = {
+      url = "github:hercules-ci/gitignore.nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    devshell = {
+      url = "github:numtide/devshell";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = inputs @ {
+    flake-parts,
+    devshell,
+    gitignore,
+    ...
+  }:
+    flake-parts.lib.mkFlake {inherit inputs;} {
```

Comment:
Sure this isn't one indentation too many? It doesn't cleanly unwind if you look at the end of the file.

Reply:
I'm not sure -- I'm still learning the Nix language and any coding style conventions that come with it.

I based the code on this example: https://github.com/brianmcgee/nix-gradle-sample/blob/main/flake.nix and have not changed the indentation. Note that there is one other place in the code with a `}:` and both are indented.